js/user-form.js

```diff
@@ -0,0 +1,134 @@
+const linkedin_id = document.getElementById('linkedin_id');
+const twitter_id = document.getElementById('twitter_id');
+const btn = document.getElementById('submission');
+
+//Error Handler
+function showError(input, message) {
+  const formGroup = input.parentElement;
+  formGroup.className = 'form-group error';
+  const small = formGroup.querySelector('small');
+  small.innerText = message;
+}
+
+//Success Handler
+function showSuccess(input) {
+  const formGroup = input.parentElement;
+  formGroup.className = 'form-group success';
+}
+
+// Username Validator
+function checkUsername(input) {
+  //Check presence of first_name
+  if (!input.value.trim().startsWith(first_name.value.toLowerCase())) {
+    showError(
+      username,
+      'Username must start with first name. Usage of funky words should be avoided.'
+    );
+  } else {
+    showSuccess(username);
+  }
+
+  //Check if username pertains to the appropriate format
+  let pattern = new RegExp(/^[a-zA-Z-]+$/g);
+  if (pattern.test(input.value.trim())) {
+    showSuccess(input);
+  } else {
+    showError(
+      username,
+      'Only alphabets and "-" are allowed. Numbers are prohibited.'
+    );
+  }
+}
+
+//Email Validator
+function checkEmail(input) {
+  const re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  if (re.test(input.value.trim())) {
+    showSuccess(input);
+  } else {
+    showError(input, 'Email is not valid');
+  }
+}
+
+//Phone Validator
+function checkPhone(input) {
+  const re = /^\(?([0-9]{3})\)?[-. ]?([0-9]{3})[-. ]?([0-9]{4})$/;
+  if (re.test(input.value.trim())) {
+    showSuccess(input);
+  } else {
+    showError(input, 'Phone number is not valid');
+  }
+}
+
+//Input fields validator
+function checkRequired(inputArr) {
+  inputArr.forEach((input) => {
+    if (input.value.trim() === '') {
+      showError(input, 'Field cannot be empty.');
+      btn.disabled = true;
+    } else {
+      showSuccess(input);
+      btn.disabled = false;
+      btn.style.opacity = 1;
+    }
+  });
+}
+
+//Form submission handler
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  checkUsername(username);
+  checkEmail(email);
+  checkPhone(phone);
+  checkRequired([
+    username,
+    first_name,
+    last_name,
+    yoe,
+    company,
+    designation,
+    github_id,
+    linkedin_id,
+  ]);
+
+  updateUserData();
+});
+
+let id;
+const getUser = async () => {
+  try {
+    let res = await fetch(`https://staging-api.realdevsquad.com/users/self`, {
+      method: 'GET',
+      credentials: 'include',
+    });
+    const data = await res.json();
+    id = data.id;
+    return id;
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+const updateUserData = async () => {
+  try {
+    let res = await fetch(`https://staging-api.realdevsquad.com/users/${id}`, {
+      method: 'PATCH',
+      credentials: 'include',
```

Comment:
1. What are we sending with the PATCH request? The data to be updated in the databse needs to be sent in the body of the PATCH request. Nothing is being sent in this request to be updated.
2. The updateUserData and getUser functions are being fired simultaneously/asynchronously, that's why the `id` sent in the URL of PATCH req will be `undefined` in this case. We need to have returned `id` from the getUser function before we actually do the PATCH req. The flow of the requests needs to be proper otherwise the form won't work as expected. Please test on your local machine before you push it so ensure that the functionality is proper.

I hope I'm making sense 😅
Please let me know if I'm confusing you.